src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1021,8 +1025,34 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     scan.Skip(method.Length);
                 }
 
-                scan.Take();
+                scan.Take(); // consume space
+
+                // begin consuming request-target
                 begin = scan;
+
+                var targetBegin = scan;
+
+                if (targetBegin.Peek() != ByteForwardSlash)
+                {
+                    string requestUriScheme;
+                    if (scan.GetKnownHttpSchema(out requestUriScheme))
```

Comment:
The scheme string wouldn't need to be allocated even if we supported unknown schema. The out parameter could be the length of the schema string since the schema from the target isn't stored anywhere.

Better yet, this could be like `GetKnownVersion` and advance the iterator itself. I see that `GetKnownMethod` doesn't advance the iterator. I would change that too for consistency.